### PR TITLE
[SPARK-30953][SQL] InsertAdaptiveSparkPlan should apply AQE on child plan of write commands

### PR DESCRIPTION
**Pull request by Ngone51**
### What changes were proposed in this pull request?

This PR changes `InsertAdaptiveSparkPlan` to apply AQE on the child plan of V1/V2 write commands rather than the command itself.

### Why are the changes needed?

Apply AQE on write commands with child plan will expose `LogicalQueryStage` to `Analyzer` while it should hider under `AdaptiveSparkPlanExec` only to avoid unexpected broken. 

### Does this PR introduce any user-facing change?

No.

### How was this patch tested?

Pass Jenkins.

### CODE REVIEW

**Comment by Ngone51** [comment]
Jenkins, retest this please.

**Comment by Ngone51** [comment]
Ping @cloud-fan please take a look, thanks!

File: sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/InsertAdaptiveSparkPlan.scala [hunk 1]
```diff
@@ -44,7 +45,7 @@ case class InsertAdaptiveSparkPlan(
 
   private def applyInternal(plan: SparkPlan, isSubquery: Boolean): SparkPlan = plan match {
     case _ if !conf.adaptiveExecutionEnabled => plan
-    case _: ExecutedCommandExec => plan
+    case _: ExecutedCommandExec | _: V2CommandExec => plan
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/InsertAdaptiveSparkPlan.scala]: ideally this is corrected, but then we need to apply AQE for input queries of command. e.g. CTAS, INSERT

**Comment by Ngone51** [comment]
ping @cloud-fan @maryannxue

**Comment by maryannxue** [comment]
You need tests to ensure future changes don't break this.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala [hunk 2]
```diff
@@ -899,4 +900,16 @@ class AdaptiveQueryExecSuite
...
+
+  test("SPARK-30953: InsertAdaptiveSparkPlan should apply AQE on child plan of write commands") {
+    withSQLConf(SQLConf.ADAPTIVE_EXECUTION_ENABLED.key -> "true") {
+      spark.range(10).repartition(1).selectExpr("id as a", "id * 2 as b")
+        .createOrReplaceTempView("t0")
+      val df = spark.sql("CREATE TABLE t1 AS SELECT a FROM t0")
+      val cmd = collect(df.queryExecution.executedPlan) {
+        case cmd: DataWritingCommandExec => cmd
+      }.head
+      assert(cmd.child.isInstanceOf[AdaptiveSparkPlanExec])
```
Comment (maryannxue) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala:912]: nit: can we also assert there is no adaptive node above the command node?

**Review by maryannxue — APPROVED** [review]
LGTM except one minor comment.

File: sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala [hunk 3]
```diff
@@ -899,4 +900,16 @@ class AdaptiveQueryExecSuite
         checkPlanDescription(mode, expected)
     }
   }
+
+  test("SPARK-30953: InsertAdaptiveSparkPlan should apply AQE on child plan of write commands") {
+    withSQLConf(SQLConf.ADAPTIVE_EXECUTION_ENABLED.key -> "true") {
+      spark.range(10).repartition(1).selectExpr("id as a", "id * 2 as b")
+        .createOrReplaceTempView("t0")
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala:907]: wrap the test with `withTempView`

File: sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala [hunk 4]
```diff
@@ -899,4 +900,16 @@ class AdaptiveQueryExecSuite
...
+      val df = spark.sql("CREATE TABLE t1 AS SELECT a FROM t0")
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala:908]: we can call `sql(....)` directly.

Comment (cloud-fan) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala:908]: for simplicity, how about `sql("CREATE TABLE t1 AS SELECT 1 col")`?

File: sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala (hunk 2 above)
Comment (cloud-fan) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala:912]: we can check the root node is the CTAS command and its child is `AdaptiveSparkPlanExec`

File: sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala (hunk 4 above)
Comment (Ngone51) [reply sql/core/src/test/scala/org/apache/spark/sql/execution/adaptive/AdaptiveQueryExecSuite.scala:908]: ok.

**Review by cloud-fan — APPROVED** [review]

**Comment by cloud-fan** [comment]
retest this please

**Comment by cloud-fan** [comment]
retest this please

**Comment by cloud-fan** [comment]
thanks, merging to master/3.0!

**Comment by Ngone51** [comment]
thanks all!

**Pull request closed without merging**